### DISCUSSION TITLE
Mikrotik CHR

### DISCUSSION THREAD

**Discussion by buczi11** [tteck/Proxmox#439] [Ideas]
Hi,
ubiquiti and tp-link omada scripts are now present. I am curious if there is a possibility and if there are many takers for Mikrotik CHR.

[Cloud Host Routed](https://wiki.mikrotik.com/wiki/Manual:CHR)

**Comment by tteck** [comment]
Would you like me to create a Mikrotik RouterOS VM script to play with?

I know nothing about running the software other than you have to use the VM console to get started. So, no support.
https://help.mikrotik.com/docs/display/ROS/First+Time+Configuration

**Comment by tteck** [comment]
@MrEddX  just a test
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/mikrotik-routeros-v4.sh)"
```

**Reply by MrEddX** [reply]
Today I finally had a chance to test it and I must say that this script is a great convenience! Yes true, it is usable for 24 hours /license key is required/, but for testing some scenarios it is more than enough. However, if more time is needed, run the script again and import the router's configuration from the previous installation. In other words, for 3 minutes of work we get new 24 hours to test and learn. 

>By the way, there is a way to access the router through a graphical environment besides the console. You have to download the [Winbox ](https://mt.lv/winbox64) application.

- _Screenshot: Accessing the VM created by the script using Winbox._

![VM-Mikrotik](https://user-images.githubusercontent.com/66828538/185763743-f42c2da2-ade3-42f2-877b-ae1d360f4245.JPG)

So Yeah, this script just works!

**Thank You!**

**Reply by tteck** [reply]
I'm glad it works. A  license level 4 at $45 doesn't seem to be a bad deal?

Should I add it to the README.md so that it's public?

**Reply by MrEddX** [reply]
> A license level 4 at $45 doesn't seem to be a bad deal?

Not at all, Router OS is a brilliant piece of software!

> Should I add it to the README.md so that it's public?

Absolutely.

**Reply by tteck** [reply]
> WebFig is a web based RouterOS utility which allows you to monitor, configure and troubleshoot the router.

Does this work?

**Reply by buczi11** [reply]
Sometimes a person won't think what he writes ;) 
There is a free trial 60 days, but in version 7.4.1 stable I have a different License window in WinBox, and in cli commands do not work

**Reply by MrEddX** [reply]
> > WebFig is a web based RouterOS utility which allows you to monitor, configure and troubleshoot the router.
> 
> Does this work?

Yes. Just fire up your browser and point to the router's IP address.

**Comment by Koliera** [comment]
Respect to you guys. I had a chance to play with a couple routerboards and the OS was just kicking my head in. Great if you want to learn but a bit too much for me :D In the end just flashed OpenWRT on them.

**Reply by MrEddX** [reply]
@Koliera Yeah. The learning curve is a bit steep at first, but then it pays off many times over. You're right for an average home user it is a bit too much. But hey, we are not average home users. That's why we are here! Right?

**Reply by Koliera** [reply]
> @Koliera Yeah. The learning curve is a bit steep at first, but then it pays off many times over. You're right for an average home user it is a bit too much. But hey, we are not average home users. That's why we are here! Right?

Yeahhh, sure....... :D
